docker/docker_client.go

```diff
@@ -224,8 +231,8 @@ func (c *dockerClient) makeRequestToResolvedURL(method, url string, headers map[
 	if c.ctx != nil && c.ctx.DockerRegistryUserAgent != "" {
 		req.Header.Add("User-Agent", c.ctx.DockerRegistryUserAgent)
 	}
-	if c.wwwAuthenticate != "" && sendAuth {
-		if err := c.setupRequestAuth(req); err != nil {
+	if sendAuth {
+		if err := c.setupRequestAuth(req, c.scope.remoteName, c.scope.actions); err != nil {
```

Comment:
`c.scope`* does not have to be a parameter, the method has direct access to `c`.